### ISSUE TITLE
[BUG] VideoCapture.Start is very CPU intensive

### ISSUE THREAD

**Issue by Voileexperiments** [emgucv/emgucv#940]
**OS / Platform**
Windows

**.Net version**
.NetFramework 4.7

**CPU Architecture**
x86_64

** Emgu CV package used**
libemgucv-windesktop-4.9.0.5494

**Describe the bug**
I was trying to capture frames periodically from a laptop camera, at a rate much lower than the FPS of the camera (let's say, the camera is 30fps and I want to grab a frame every second). I can't set the camera FPS as it is fixed.

I already verified that this is doable with a basic Python loop, with minimal CPU usage:

```python
import cv2
import time

vid = cv2.VideoCapture(0, cv2.CAP_DSHOW)
while(True):
    ret, frame = vid.read()
    cv2.imshow('frame', frame)
    if cv2.waitKey(1) & 0xFF == ord('q'): break
    time.sleep(1.0)

vid.release()
cv2.destroyAllWindows()
```

With Emgu.CV, I can do this:

```csharp
var videoCapture = VideoCapture(0, VideoCapture.API.DShow);
VideoCapture.Start();
```

Then call `VideoCapture.Retrieve` on the separate thread every second. However, this also saturates 2 of my logical cores, which is undesirable.

`.Start()` is not an OpenCV native function, so this is not an OpenCV issue, it has to do with Emgu.CV itself.

---

Yes, it is possible to do the following alternatives, but they're not ideal:

- Call `VideoCapture.QueryFrame`/`VideoCapture.Read` every time I want to get a image

This would copy the Python code's behaviour. This is also known to *not* return latest frame when frame buffer exists, and since I'm reading image at a slow rate, this might mean the actual image I get lags from the current time.

- Create a separate thread to repeatedly call `VideoCapture.Grab` at a interval (e.g 0.1s), then use `VideoCapture.Retrieve` to get a image

This would just be a poor man's imitation of the current `VideoCapture.Start()`, with a worse handling of grab readiness that change over time (`VideoCapture.Start` does this by actually inspecting grab state).

**Comment by emgucv** [comment]
A direct translation will be this:


```
using System;
using System.Threading;
using Emgu.CV;
using Emgu.CV.CvEnum;
using Emgu.CV.Structure;

namespace VideoCaptureTest
{
    class Program
    {
        static void Main(string[] args)
        {
            using (VideoCapture vid = new VideoCapture(0, VideoCapture.API.DShow))
            {
                while (true)
                {
                    using (Mat frame = new Mat())
                    {
                        vid.Read(frame);
                        CvInvoke.Imshow("frame", frame);
                        if ((CvInvoke.WaitKey(1) & 0xFF) == 'q') 
                            break;
                    }
                    Thread.Sleep(1000); //1000 milliseconds == 1 sec
                }
            }

            CvInvoke.DestroyAllWindows(); //Destroy all windows if key is pressed
        }
    }
}
```

VideoCapture.Start() do not try to sleep in-between the captures, it will try to grab with the highest frame rate as possible. If you need to sleep for 1 sec after every capture, you can translate your code directly into C#.

**Comment by emgucv** [comment]
The Start() function was doing what the documentation says:

https://www.emgu.com/wiki/files/4.9.0/document/html/M_Emgu_CV_VideoCapture_Start.htm

> Start the grab process in a separate thread. Once started, use the ImageGrabbed event handler and RetrieveGrayFrame/RetrieveBgrFrame to obtain the images.

This implement a very simple loop for VideoCapture. A very common video capture logic is to capture frames and process them as soon as it is possible. Once it is done, capture another one.

You are free to add any processing logic to the ImageGrabbed event, including the logic to read a single frame and sleep until it is one second from the last time the previous frame is captured. The function is extensible. You can add your own logic in the ImageGrabbed event for your need. 

You are also free to use the existing Open CV syntax from the response above.

**Comment by emgucv** [comment]
Closing this with the recommended patterns, since `Start()` behaves as documented — it is deliberately a tight grab-process-repeat loop, which is the right default for "process every frame as fast as possible" workloads. For low-rate sampling there are two supported approaches, both avoiding the CPU saturation:

1. **Throttle the grab loop itself.** The `ImageGrabbed` event is invoked *synchronously* from the grab thread, so a delay inside the handler paces the whole loop:
```csharp
capture.ImageGrabbed += (s, e) =>
{
    capture.Retrieve(frame);
    // ... process frame ...
    Thread.Sleep(1000);   // paces the grab loop; CPU idles between frames
};
capture.Start();
```
This keeps `Start()`'s grab-state management while sampling at your rate, and because a grab immediately precedes each retrieve, the frame is current.

2. **Timer-based `Read` with a minimal buffer.** The stale-frame concern you raised about slow `Read` calls is addressed by shrinking the driver's frame queue: `capture.Set(CapProp.Buffersize, 1);` (honored by the DShow/MSMF backends) makes each `Read` return the newest frame, making a simple one-second timer loop equivalent to your Python snippet.

Closing as answered — if you measure high CPU *with* the throttled handler on a current release, please open a new issue with the measurement details.
